### PR TITLE
Explore: make sure panes are initialized in order

### PR DESCRIPTION
**Pull request by Elfo404**
**What is this feature?**

Ensures panes in Explore are initialized in the same order as defined in the URL. Given how we lazy load plugins (or in general because of the async nature of the init process) it may have happened that loading a ds on the right pane finished before than loading the left pane one, which caused Explore to switch the panes in the state.

**Why do we need this feature?**

Panes ordering should be consistent across reloads

**Which issue(s) does this PR fix?**:

Fixes #

**Special notes for your reviewer:**

Not backporting this as this behaviour was not present in 10.0.x

Please check that:

- [ ] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.

**How to test it:**
Reproducing the issue in main is quite easy: open Explore with 2 panes, on the left pick Loki as a datasource (Prometheus or Elastic would work as well), on the right pick testdata and reload the page. without this fix the panes will be swapped.

### CODE REVIEW

File: public/app/features/explore/hooks/useStateSync/index.ts [hunk 1]
```diff
@@ -235,16 +233,14 @@ export function useStateSync(params: ExploreQueryParams) {
...
         initState.current = 'done';
-
         // we need to use partial here beacuse replace doesn't encode the query params.
         location.partial(
           {
             // partial doesn't remove other parameters, so we delete all the current one before adding the new ones.
-            ...Object.keys(location.getSearchObject()).reduce<Record<string, unknown>>((acc, key) => {
-              acc[key] = undefined;
-              return acc;
+            ...Object.keys(location.getSearchObject()).reduce<Record<string, unknown>>((acc_1, key) => {
```
Comment (ifrost) [inline public/app/features/explore/hooks/useStateSync/index.ts:241]: why `acc_1`?

**Review by ifrost — APPROVED** [review]
Pretty nasty bug :O Thank you for catching it before anyone else :D

File: public/app/features/explore/hooks/useStateSync/index.ts (hunk 1 above)
Comment (Elfo404) [reply public/app/features/explore/hooks/useStateSync/index.ts:241]: because vscode does funny things when autorefactoring, will change

Comment (Elfo404) [reply public/app/features/explore/hooks/useStateSync/index.ts]: 5851700b3a :D

**Pull request merged**